app/lib/dns.server.ts

```diff
@@ -0,0 +1,203 @@
+    }
+
+    if (!isValueValid(type, value)) {
+      throw new Error('Invalid value provided');
+    }
+
+    const command = new ChangeResourceRecordSetsCommand({
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: 'UPSERT',
+            ResourceRecordSet: {
+              Name: name,
+              Type: type,
+              ResourceRecords: [
+                {
+                  Value: value,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+    const response: ChangeResourceRecordSetsResponse = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error(`Missing ID in ChangeInfo`);
+    }
+    return response.ChangeInfo.Id;
+  } catch (error) {
+    if (error instanceof Error) {
+      // TODO: fix with better logging...
+      console.warn(error);
+      throw new Error(`Error occurred while updating resource record: ${error.message}`);
+    }
+  }
+};
+
+export const deleteRecord = async (
+  hostedZoneId: string,
+  type: RecordType,
+  name: string,
+  value: string
+) => {
+  try {
+    if (!isNameValid(name)) {
+      throw new Error('Invalid name provided');
+    }
+
+    if (!isValueValid(type, value)) {
+      throw new Error('Invalid value provided');
+    }
+
+    const command = new ChangeResourceRecordSetsCommand({
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: 'DELETE',
+            ResourceRecordSet: {
+              Name: name,
+              Type: type,
+              ResourceRecords: [
+                {
+                  Value: value,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+
+    const response: ChangeResourceRecordSetsResponse = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error('Missing ID in ChangeInfo');
+    }
+    return response.ChangeInfo.Id;
+  } catch (error) {
+    if (error instanceof Error) {
+      // TODO: fix with better logging...
+      console.warn(error);
+      throw new Error(`Error occurred while deleting resource record: ${error.message}`);
+    }
+  }
+};
+
+export const getChangeStatus = async (changeId: string) => {
+  try {
+    const command = new GetChangeCommand({
+      Id: changeId,
+    });
+    const response: GetChangeResponse = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Status) {
+      throw new Error('Could not get ChangeIno for requested ID');
+    }
+    return response.ChangeInfo.Status;
+  } catch (error) {
+    if (error instanceof Error) {
+      // TODO: fix with better logging...
+      console.warn(error);
+      throw new Error(`Error occurred while getting change status: ${error.message}`);
+    }
+  }
+};
+
+const isNameValid = (name: string) => {
+  return /^[a-z0-9-]+.[a-z]+.[a-z]+$/.test(name);
```

Comment:
It's valid if you include the trailing `.` as well (i.e., canonical name version).  Add a `\.?$` to the end